packages/s3-request-presigner/src/presigner.spec.ts

```diff
@@ -87,4 +87,19 @@ describe("s3 presigner", () => {
       [EXPIRES_QUERY_PARAM]: "900",
     });
   });
+
+  it("should disable hoisting server-side-encryption headers to query", async () => {
+    const signer = new S3RequestPresigner(s3ResolvedConfig);
+    const signed = await signer.presign({
+      ...minimalRequest,
+      headers: {
+        ...minimalRequest.headers,
+        "x-amz-server-side-encryption": "kms",
```

Comment:
Small suggestion: also test other headers, ones prefixed with `x-amz-server-side-encryption` to make sure all cases are covered, e.g. `x-amz-server-side-encryption-aws-kms-key-id`.